internal/blueprint/repository_customizations.go

```diff
@@ -326,3 +330,84 @@ func validateCustomRepository(repo *RepositoryCustomization) error {
 	}
 	return nil
 }
+
+// DeepCopy returns a deep copy of the given slice of RepositoryCustomization
+// Since we are replacing the gpgkeys of the customization with the file path
+// we don't want to modify the original slice.
+func DeepCopyCustomRepositories(repos []RepositoryCustomization) []RepositoryCustomization {
```

Comment:
I don't think this is needed. My suggestion would be to:

Merge `GetInlineGPGKeysFromCustomRepositories()` and `RepoCustomizationsToRepoConfig()` into a single function, which returns `(map[string][]rpmmd.RepoConfig, []*fsnode.File, error)`. In a single iteration, you create a new `rpmmd.RepoConfig` which already has deep copy of all values from the repository customization structure. You would then generate a `*fsnode.File` for the repo if the GPG key was provided inline and you would modify the `rpmmd.RepoConfig` to point to a file, before adding it to the map of slices of repo configs. This should not create any issues (AFAICT).

Doing the conversion to internal representations in two steps is the only reason that you need this helper function and it also creates possibility to call them in the wrong order and not have inline GPG keys converted to file paths. Modifying the `rpmmd.RepoConfig` after conversion in a single function will eliminate any need for deep copy since you won't ever modify the customization structure.

Also I would appreciate if you could add unit tests for all of these newly added helper functions.

Reply:
Okay, cool sounds good to me. I will try merge the functions like you suggest and yes, absolutely, I'll add unit tests... just wanted to make sure I was on the right track before doing that :)

Reply:
You are definitely on the right track, you got it right! My suggestion would just simplify the implementation, but the resulting behavior should stay the same 😉